### DISCUSSION TITLE
Is there a native way to do lazy loading in Kit?

### DISCUSSION THREAD

**Discussion by frederikhors** [sveltejs/kit#4493] [Q&A]
What do you think about github.com/kaisermann/svelte-loadable?

Is there a native way to do that in Kit?

**Comment by frederikhors** [comment]
Is there anyone who can answer?

**Comment by david-plugge** [comment]
Can't u just use a svelte await block? 

```svelte
{#await import('./MyComponent.svelte')}
  Loading... 
{:then { default: component } }
  <svelte:component this={component} />
{:catch error} 
  Failed to load component
{/await}
```

**Reply by frederikhors** [reply]
Is this lazy? I mean is the import called only when needed?

**Reply by david-plugge** [reply]
Yes, if you wrap it in an if statement

**Reply by karimfromjordan** [reply]
Just for reference, dynamic imports are a feature of Vite/Rollup and documented in the Vite docs. It also supports variables: https://vitejs.dev/guide/features.html#dynamic-import

**Reply by frederikhors** [reply]
And how can we use this lazily in Kit?
